pkg/controllers/machinesetsync/machineset_sync_controller.go

```diff
@@ -98,88 +105,218 @@ func (r *MachineSetSyncReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return nil
 }
 
-// Reconcile reconciles CAPI and MAPI machines for their respective namespaces.
-//
-//nolint:funlen
+// Reconcile reconciles CAPI and MAPI MachineSets for their respective namespaces.
 func (r *MachineSetSyncReconciler) Reconcile(ctx context.Context, req reconcile.Request) (ctrl.Result, error) {
-	logger := log.FromContext(ctx, "namespace", req.Namespace, "name", req.Name)
+	logger := log.FromContext(ctx).WithValues("namespace", req.Namespace, "name", req.Name)
```

Comment:
Could be useful to tell also the resource type (e.g. machineset or machine), as the log lines might intertwine

Reply:
We get this for free somehow that I don't entirely understand right now: 

```
E1028 17:15:16.233708   14061 controller.go:316] "Reconciler error" err="failed to get CAPI InfraCluster: AWSCluster.infrastructure.cluster.x-k8s.io \"\" not found" controller="machineset" controllerGroup="machine.openshift.io" controllerKind="MachineSet" MachineSet="openshift-machine-api-rk7pk/foo" namespace="openshift-machine-api-rk7pk" name="foo" reconcileID="3d65de5d-360d-4cb2-b835-b3c7b3c82a15"
```